.circleci/config.yml

```diff
@@ -1,44 +1,90 @@
-version: 2
-defaults: &defaults
-  working_directory: /home/circleci/oce
-jobs:
-  build:
-    <<: *defaults
+version: 2.1
+
+executors:
+  build-executor:
     docker:
-      - image: circleci/openjdk:8-jdk-browsers
+      - image: opennms/build-env:1.0-b4
+        environment:
+          MAVEN_OPTS: -Xmx2g
 
-    environment:
-      MAVEN_OPTS: -Xmx2g
+  docs-executor:
+    docker:
+      - image: antora/antora:2.0.0
 
-    steps:
-      - run:
-          name: Install RPM
-          command: sudo apt-get update; sudo apt-get install rpm
+  netlify-cli-executor:
+    docker:
+      - image: opennms/netlify-cli:2.8.3-b1
 
-      # Restore source cache
-      - restore_cache:
-          keys:
-            - source-v1-{{ .Branch }}-{{ .Revision }}
-            - source-v1-{{ .Branch }}-
-            - source-v1-
+  package-cloud-cli-executor:
+    docker:
+      - image: opennms/package-cloud-cli:0.3.05-b1
+
+workflows:
+  build-deploy:
+    jobs:
+      # Run build jobs for all branches and any tag
+      - pre-build:
+          filters:
+            tags:
+              only: /.*/
+      - build:
+          requires:
+            - pre-build
+          filters:
+            tags:
+              only: /.*/
+      - build-docs:
+          requires:
+            - pre-build
+          filters:
+            tags:
+              only: /.*/
 
+      # Run deploy jobs just on version tags and ignore branches
```

Comment:
We still want to deploy snapshot artifacts for the "master" branch (as we do currently)